x/claim/keeper/msg_server_claim_eth.go

```diff
@@ -57,13 +57,28 @@ func (k msgServer) ClaimEth(goCtx context.Context, msg *types.MsgClaimEth) (*typ
 		Address:                msg.Creator,
 		Chain:                  types.ARKEO,
 		InitialClaimableAmount: ethClaim.InitialClaimableAmount,
-		ActionCompleted:        []bool{false, false},
+		ActionCompleted:        []bool{true, false, false}, // true so they do not make them claim again on arkeo.
 	}
+
+	ctx.EventManager().EmitEvents(sdk.Events{
```

Comment:
emit event should be moved to the end , since at this stage , no fund has been credit to user's account , if could error out after this step , then we might end up a scenario user didn't get fund , but event is on chain

Reply:
nope , for example , if line 71 fail , then the event will be write to block , however the user didn't get anything

Reply:
> PostHandlers: [PostHandler](https://docs.cosmos.network/main/core/baseapp#posthandler)s run after the execution of the message. If they fail, the state change of runMsgs, as well of PostHandlers, are both reverted.

This seems contradictory to the documentation.  If a transaction errors, and the state is not reverted, we have other logical problems that need to be dealt with and this behavior would be quite odd.

Reply:
I will create a ticket for further investigation into this, because its problematic if these transaction are not atomic as you suggest.

Additionally, we have a different issue here as well where a user could clobber an existing claim record if trying to assign to an arkeo address that already has a drop to claim. 

Will create issues for both of these and we can address in the future while keeping this moving forward.

Reply:
https://github.com/arkeonetwork/arkeo/issues/43
https://github.com/arkeonetwork/arkeo/issues/42

Reply:
Looking at the source code again, I think you are right , if we return error , then the state will not be changed.